### PR TITLE
Refactor drive handler logic

### PR DESCRIPTION
**Pull request by ashmrtn**
Do a couple of high-level things with drive handlers:
* pull out generic functionality into a base handler
* use handler-specific implementations of AugmentItemInfo
* remove old switch-based augmentItemInfo

Overall goal is to reduce places we use service type
comparisons to handle behavior differences

---

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [x] :broom: Tech Debt/Cleanup

#### Issue(s)

* #4254

#### Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/m365/collection/drive/handler_utils.go [hunk 1]
```diff
@@ -5,96 +5,37 @@ import (
...
-			ItemName:   ptr.Val(item.GetName()),
-			ItemType:   details.SharePointLibrary,
-			Modified:   ptr.Val(item.GetLastModifiedDateTime()),
-			Owner:      creatorEmail,
-			ParentPath: pps,
-			SiteID:     resource.ID(),
-			Size:       size,
-			WebURL:     resource.Name(),
-		}
+	return *ed.(*string)
```
Comment (ryanfkeepers) [inline src/internal/m365/collection/drive/handler_utils.go:29]: If you'd be so kind.
```suggestion
        // TODO(keepers): str package with fallbacks.
	return *ed.(*string)
```

File: src/internal/m365/collection/drive/item_handler.go [hunk 2]
```diff
@@ -21,16 +22,68 @@ import (
 // backup
 // ---------------------------------------------------------------------------
 
+type baseItemHandler struct {
+	ac api.Drives
+}
+
+func (h baseItemHandler) NewDrivePager(
+	resourceOwner string, fields []string,
```
Comment (ryanfkeepers) [inline src/internal/m365/collection/drive/item_handler.go:30]:
```suggestion
	resourceOwner string,
	fields []string,
```

File: src/internal/m365/collection/drive/item_handler.go [hunk 3]
```diff
@@ -21,16 +22,68 @@ import (
 // backup
 // ---------------------------------------------------------------------------
 
+type baseItemHandler struct {
```
Comment (ryanfkeepers) [inline src/internal/m365/collection/drive/item_handler.go:25]: This name (along with the other "itemFoo" names) is out of date, and was originally chosen to represent the file name it came from.  Would you mind making this change as long as you're in the refactor?
```suggestion
type baseUserDriveHandler struct {
```

File: src/internal/m365/collection/drive/library_handler.go [hunk 4]
```diff
@@ -17,10 +18,54 @@ import (
 	"github.com/alcionai/corso/src/pkg/services/m365/api"
 )
 
+type baseLibraryHandler struct {
```
Comment (ryanfkeepers) [inline src/internal/m365/collection/drive/library_handler.go:21]: Similar request here with naming.
```suggestion
type baseSiteDriveHandler struct {
```

**Review by ryanfkeepers — APPROVED** [review]
Looks great, thank you for the refactor!

File: src/internal/m365/collection/drive/item_handler.go (hunk 3 above)
Comment (ashmrtn) [reply src/internal/m365/collection/drive/item_handler.go:25]: I'll spin up another PR for this and other naming changes

**Pull request merged**